File: coding_interviews/leetcode/easy/ransom_note/ransom_note.py
def can_construct(ransom_note, magazine):
    if ransom_note == '':
        return True

    if magazine == '':
        return False

    letters_counter = {}

    for letter in magazine:
        if letter in letters_counter:
            letters_counter[letter] += 1
        else:
            letters_counter[letter] = 1

    for char in ransom_note:
        if char not in letters_counter or letters_counter[char] == 0:
            return False

        letters_counter[char] -= 1

    return True

File: coding_interviews/leetcode/easy/ransom-note/ransom-note.js
function canConstruct(ransomNote, magazine) {
  let map = {};

  for (let char of magazine) {
    if (map[char]) {
      map[char] += 1;
    } else {
      map[char] = 1;
    }
  }

  for (let char of ransomNote) {
    if (map[char]) {
      map[char]--;
    } else {
      return false;
    }
  }

  return true;
}
